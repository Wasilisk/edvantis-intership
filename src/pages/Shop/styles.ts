import styled from "styled-components";

export const ShopStyles = styled.div`
  max-width: 1200px;
  margin: 50px auto;
  padding: 0 24px;

  
  .title {
    font-size: 30px;
    line-height: 1.2;
    font-weight: 600;
    color: #333333;
    padding-left: 20px;
    margin-bottom: 20px;
  }
  
  .product-grid {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-row-gap: 40px;
  }

  @media (max-width: 1024px) {
    .product-grid {
      grid-template-columns: repeat(3, 1fr);
    }
  }
  @media (max-width: 768px) {
    .title {
      text-align: center;
      padding: 0;
    }
    .product-grid {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  @media (max-width: 425px) {
    .product-grid {
      grid-template-columns: repeat(1, 1fr);
    }
  }
`;