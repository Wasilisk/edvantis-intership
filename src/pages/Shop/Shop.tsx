import React, {useEffect} from 'react';
import {useAppDispatch, useAppSelector} from "../../hooks";
import {selectAllProducts} from "../../store/slices/Products/selectors";
import {getAllProducts} from "../../store/slices/Products";
import ProductCard from "../../components/Cards/ProductCard";
import {ShopStyles} from "./styles";

const Shop = () => {
    const dispatch = useAppDispatch();
    const allProducts = useAppSelector(selectAllProducts);

    useEffect(() => {
        dispatch(getAllProducts());
    }, [])

    return (
        <ShopStyles>
            <h4 className="title">Products ({allProducts?.length})</h4>
            <div className="product-grid">
                {
                    allProducts && allProducts.map(product => <ProductCard
                            key={product.id}
                            buttonVariant="filled"
                            {...product}
                        />
                    )
                }
            </div>
        </ShopStyles>
    );
};

export default Shop;