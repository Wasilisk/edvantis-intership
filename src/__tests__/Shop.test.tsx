import {render} from "./__utils__/store-wrapper";
import Shop from "../pages/Shop/Shop";
import {mockProductCard} from "./__mock__/mock-product-card";

describe('Shop component', () => {
    it('should render header component', () => {
        const {asFragment, getByText, getAllByTestId} = render(<Shop/>, {
            initialState: {
                products: {
                    allProducts: [...Array(15)].map((_, index) => Object.assign({}, mockProductCard, { id: index }))
                }
            }

        });

        const title = getByText('Products (15)');
        expect(title).toBeDefined();

        const productCards = getAllByTestId('product-card');
        expect(productCards.length).toBe(15);

        expect(asFragment()).toMatchSnapshot();
    })
})