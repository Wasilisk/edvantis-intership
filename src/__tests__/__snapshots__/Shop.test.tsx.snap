// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackSection component should render header component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf jIYOak"
  >
    <h4
      class="title"
    >
      Products (15)
    </h4>
    <div
      class="product-grid"
    >
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-bczRLJ eZuxEp"
        data-testid="product-card"
      >
        <a
          class="card-image"
          href="/shop/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/shop/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-gsnTZi bxYKFi"
          type="button"
        >
          Order Now
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
